### PR TITLE
feat: integrate export with stateful api(issue #640, #594)

### PR DESCRIPTION
**Pull request by denys-kolomiitsev**
**Issues:**

- #640 - export part;
- #594

### CODE REVIEW

File: apps/chat/src/store/prompts/prompts.epics.ts [hunk 1]
```diff
@@ -322,32 +325,60 @@ const deleteFolderEpic: AppEpic = (action$, state$) =>
...
+      }
+
+      const appName = SettingsSelectors.selectAppName(state$.value);
+
       exportPrompt(
         prompt,
         PromptsSelectors.selectParentFolders(state$.value, prompt.folderId),
+        appName,
       );
+      //TODO create export success action
```
Comment (mikitabut) [inline apps/chat/src/store/prompts/prompts.epics.ts]: TODO

**Review by mikitabut — DISMISSED** [review]

File: apps/chat/src/store/prompts/prompts.epics.ts (hunk 1 above)
Comment (denys-kolomiitsev) [reply apps/chat/src/store/prompts/prompts.epics.ts]: No need to implement for now. TODO removed.

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-665.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-665.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/store/import-export/importExport.epics.ts [hunk 2]
```diff
@@ -167,14 +254,46 @@ const importConversationsEpic: AppEpic = (action$, state$) =>
...
 
-      return concat(
-        of(ImportExportActions.importConversationsSuccess()),
-        of(
-          ConversationsActions.importConversationsSuccess({
-            conversations: history,
-            folders,
-          }),
+      const conversationsInfoToUpload = history.filter((conversation) => {
+        return conversation.status !== 'LOADED';
```
Comment (Alexander-Kezik) [inline apps/chat/src/store/import-export/importExport.epics.ts]:
```suggestion
        return conversation.status !== 'LOADED';
```

```suggestion
        return conversation.status !== UploadStatus.LOADED;
```
Don't forget to import this

File: apps/chat/src/store/prompts/prompts.epics.ts [hunk 3]
```diff
@@ -332,48 +338,101 @@ const deleteFolderEpic: AppEpic = (action$, state$) =>
...
+        promptsListing: PromptService.getPrompts(undefined, true), //listing of all entities
+        promptsHistory: of(payload.promptsHistory),
+      }),
+    ),
+    switchMap(({ promptsListing, promptsHistory }) => {
+      let foldersIds: (string | undefined)[] = [];
+      let folders: FolderInterface[] = [];
+      let currentPrompts: Observable<(Prompt | null)[]> = of([]);
+
+      if (promptsListing.length) {
```
Comment (Alexander-Kezik) [inline apps/chat/src/store/prompts/prompts.epics.ts]:
```
if (promptsListing.length === 0) {
    return of({
      currentPrompts: of([]),
      currentFolders: of([]),
      promptsHistory: of(promptsHistory),
    });
  }
  
  const foldersIds = ...;
  const folders = ...;
```
  
  Maybe it's will be better to do something like this?

**Review by Alexander-Kezik — CHANGES_REQUESTED** [review]

File: apps/chat/src/store/prompts/prompts.epics.ts (hunk 3 above)
Comment (denys-kolomiitsev) [reply apps/chat/src/store/prompts/prompts.epics.ts]: Thank you for suggestions!
Changed.

File: apps/chat/src/store/import-export/importExport.epics.ts (hunk 2 above)
Comment (denys-kolomiitsev) [reply apps/chat/src/store/import-export/importExport.epics.ts]: Changed to 'UploadStatus.LOADED'!

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-665.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by Alexander-Kezik — DISMISSED** [review]

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-665.nightly-test.deltixhub.io
>E2E tests status: failed

**Review by Alexander-Kezik — APPROVED** [review]

**Pull request merged**